File: backend/src/main/java/com/lerneon/backend/services/BusinessSectorService.java
package com.lerneon.backend.services;

import com.lerneon.backend.models.entity.BusinessSector;

import java.util.List;

public interface BusinessSectorService {
    List<BusinessSector> findAllBusinessSectors();

    BusinessSector findBusinessSectorById(Integer id);

    BusinessSector createBusinessSector(BusinessSector businessSector);

    BusinessSector updateBusinessSector(Integer id, BusinessSector businessSector);

    BusinessSector deleteBusinessSector(Integer id);
}

File: client/services/business.sector.service.ts
import { ApiClient } from "@/services/config/api.client";
import { SuccessResponse } from "@/types/payload/response/common/success.response";
import { BusinessSector } from "@/types/entity/business.sector";

export class BusinessSectorService {
  static async findAllBusinessSectors(): Promise<
    SuccessResponse<BusinessSector[]>
  > {
    return ApiClient.get("business-sector").json<
      SuccessResponse<BusinessSector[]>
    >();
  }

  static async findBusinessSectorById(
    id: number,
  ): Promise<SuccessResponse<BusinessSector>> {
    return ApiClient.get(`business-sector/{id}`).json<
      SuccessResponse<BusinessSector>
    >();
  }
}
